modules/button/react/lib/IconButton.tsx

```diff
@@ -85,92 +110,161 @@ export const IconButtonCon = styled('button', {
-            backgroundColor: 'transparent',
-            ...getFillSelector(colors.blueberry400),
-            ...getAccentSelector(colors.blueberry400),
-          },
-          '&:not([disabled]):focus': {
-            ...(toggled ? focusRing(2, 0) : {}),
-          },
-          '&:hover, &:focus:hover': {
-            backgroundColor: colors.soap300,
-          },
-          backgroundColor: colors.frenchVanilla100,
-          ...getFillSelector(colors.blueberry400),
-          ...getAccentSelector(colors.blueberry400),
-        };
-      case IconButtonVariant.Circle:
-        return {
-          '&:active': {
-            ...getFillSelector(colors.blueberry400),
-            ...getAccentSelector(colors.frenchVanilla100),
-          },
-          ...getBackgroundSelector(colors.blueberry400),
-          ...getFillSelector(colors.blueberry400),
-          ...getAccentSelector(colors.frenchVanilla100),
-        };
-      case IconButtonVariant.CircleFilled:
-        return {
-          backgroundColor: colors.blueberry400,
-          '&:hover, &:focus&:hover': {
-            backgroundColor: colors.blueberry500,
-            ...getAccentSelector(colors.blueberry500),
-          },
-          ...getBackgroundSelector(colors.frenchVanilla100),
-          ...getFillSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.blueberry400),
-        };
-      case IconButtonVariant.Inverse:
-        return {
-          '&:hover span .wd-icon-fill, span .wd-icon-fill': {
-            fill: colors.frenchVanilla100,
-          },
-          ...getBackgroundSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.licorice200),
-        };
-      case IconButtonVariant.InverseFilled:
-        return {
-          backgroundColor: 'rgba(0, 0, 0, 0.4)',
-          '&:focus': {
-            backgroundColor: 'rgba(0, 0, 0, 0.4)',
-          },
-          ...getBackgroundSelector(colors.frenchVanilla100),
-          ...getFillSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.licorice200),
-          '&:active': {
-            ...getAccentSelector(colors.licorice200),
-          },
-        };
-    }
-  }
+        switch (variant) {
+          case IconButtonVariant.SquareFilled:
+          default:
+            return {
+              '&:focus&:hover, &:focus, &:active': {
+                backgroundColor: colors.blueberry400,
+                ...getFillSelector(colors.frenchVanilla100),
+                ...getAccentSelector(colors.frenchVanilla100),
+              },
+              '&:not([disabled]):focus': {
+                backgroundColor: colors.blueberry500,
+                ...(toggled ? focusRing(2, 2) : {}),
+              },
+              '&:hover': {
+                backgroundColor: colors.blueberry500,
+              },
+              ...mouseFocusBehavior({
+                '&:focus:active': {
+                  backgroundColor: `${colors.blueberry500} !important`,
```

Comment:
Unfortunately, i have to override the mouseFocusBehavior for inverse and inverse filled since we no longer have a hover or active state for these buttons. I'm open to a different approach, thoughts?